api/classes/publisher.js

```diff
@@ -201,6 +201,18 @@ class BasePublisher {
     });
   }
 
+  updateProjectReadOnlyProperty(readonly) {
+    if (typeof readonly !== `boolean`) {
+      return Promise.resolve();
+    }
+
+    return projectsQueryBuilder
+    .updateOne(this.project.id, { readonly })
+    .then(() => {
+      this.project.readonly = readonly;
```

Comment:
would it not make more sense to refetch the project, rather than assume only the `.readonly` property got updated? This is async code, after all.